### PR TITLE
Port compare page to Typescript

### PR DESCRIPTION
**Pull request by Kobzol**
This was the largest UI page to port. It was partially in Vue before, so I mostly just copied it over, refactored it slightly and cleaned up various stuff.

### CODE REVIEW

File: site/frontend/src/pages/compare/benchmarks/benchmarks.vue [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+const props = defineProps<BenchmarkProps>();
+
+function Section({title, link, linkUp}: {title: string, link: string, linkUp: boolean}) {
+  return (
+    <div class="category-title">
+      {title} benchmarks
+      <span title={`To ${link} benchmarks`}>&nbsp;
+        <a href={`#${link}-benchmarks`}>{linkUp ? "⮭" : "⮯"}</a>
```
Comment (rylev) [inline site/frontend/src/pages/compare/benchmarks/benchmarks.vue:22]: This doesn't seem to render properly in the diff?

Screenshot 2023-05-19 at 14 49 27

**Review by rylev — APPROVED** [review]
I didn't look at this super closely, but the overall shape looks good. I think this is an improvement over the status quo.

File: site/frontend/src/pages/compare/benchmarks/benchmarks.vue (hunk 1 above)
Comment (Kobzol) [reply site/frontend/src/pages/compare/benchmarks/benchmarks.vue:22]: It's fine in my source editor, probably just some visual bug.

**Comment by Kobzol** [comment]
Ok, thanks! Let's ship it.

**Pull request merged**